src/gui/search/searchtab.cpp

```diff
@@ -341,6 +398,40 @@ void SearchTab::displayToggleColumnsMenu(const QPoint&)
     }
 }
 
+void SearchTab::searchFinished(bool cancelled)
+{
+    if (cancelled)
+        setStatus(Status::Aborted);
+    else if (m_noSearchResults)
+        setStatus(Status::NoResults);
+    else
+        setStatus(Status::Finished);
+}
+
+void SearchTab::searchFailed()
+{
+    setStatus(Status::Error);
+}
+
+void SearchTab::appendSearchResults(const QList<SearchResult> &results)
+{
+    for (const SearchResult &result : results) {
+        // Add item to search result list
+        int row = m_searchListModel->rowCount();
+        m_searchListModel->insertRow(row);
+
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::NAME), result.fileName); // Name
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::DL_LINK), result.fileUrl); // download URL
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::SIZE), result.fileSize); // Size
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::SEEDS), result.nbSeeders); // Seeders
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::LEECHES), result.nbLeechers); // Leechers
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::ENGINE_URL), result.siteUrl); // Search site URL
+        m_searchListModel->setData(m_searchListModel->index(row, SearchSortModel::DESC_LINK), result.descrLink); // Description Link
```

Comment:
I know it's old code but this seems too much updates to me, is it possible to construct a `QStandardItem` with all columns filled in and then call `QStandardItemModel ::insertRow(int row, QStandardItem *item)`?

Reply:
We still need a QStandardItem list to do it. The appropriate method is `void QStandardItemModel::insertRow(int row, const QList<QStandardItem *> &items)`.
The second overload is for single-column models:
```c++
void QStandardItemModel::insertRow(int row, QStandardItem *item);
// This is an overloaded function.
// Inserts a row at row containing item.
// When building a list or a tree that has only one column, this function provides a convenient way 
// to append a single new item.
```